src/utils/vueCompatSupport.ts

```diff
@@ -0,0 +1,15 @@
+import { FindAllComponentsSelector } from '../types'
+
+export function convertLegacyVueExtendSelector(
+  selector: any
+): FindAllComponentsSelector {
+  if (
+    typeof selector === 'function' &&
+    selector.name === 'SubVue' &&
```

Comment:
I'm not super familiar with how the compat build works - what is `SubVue`? Where does this come from?

Reply:
It comes from here: https://github.com/vuejs/vue-next/blob/51d2be20386d4dc59006d31a1cc96676871027ce/packages/runtime-core/src/compat/global.ts#L229

Unfortunately there is no 100% reliable way to check that "component is coming from legacy extend" but I find `SubVue` name matching reliable enough to work in 99.9% cases (come-on, if you name your functional component `SubVue` you're in trouble :))